File: src/main/kotlin/com/codeAI/models/Task.kt
package com.codeAI.models

import io.quarkus.hibernate.reactive.panache.PanacheEntity
import org.hibernate.annotations.CreationTimestamp
import java.time.ZonedDateTime
import javax.persistence.*


@Entity
@Table(name = "tasks")
class Task : PanacheEntity() { // omitted field declarations


    @Column(nullable = false)
    var title: String? = null

    @Column(length = 1000)
    var description: String? = null
   @Column(nullable = false)
    var priority: Int? = null

    @ManyToOne(optional = false)
    var user: User? = null

    @Column(nullable = false)
    var complete: ZonedDateTime? = null

    @ManyToOne
    var project: Project? = null

    @CreationTimestamp
    @Column(updatable = false, nullable = false)
    var created: ZonedDateTime? = null

    @Version
    var version : Int? = null
}

File: frontend_angular/src/models/Task.ts
export class Task {


	constructor(id: number | undefined,
	            title: string | undefined,
	            description: string | undefined,
	            priority: number | undefined,
	            user: User | undefined,
	            complete: Date | undefined,
	            project: Project | undefined,
	            created: Date | undefined,
	            version: number | undefined) {
		this.id = id;
		this.title = title;
		this.description = description;
		this.priority = priority;
		this.user = user;
		this.complete = complete;
		this.project = project;
		this.created = created;
		this.version = version;
	}

	id: number | undefined;
	title: string | undefined;
	description: string | undefined;
	priority: number | undefined;
	user: User | undefined;
	complete: Date | undefined;
	// @ts-ignore
	project: Project | undefined;
	created: Date | undefined;
	version: number | undefined;
}


export class Project {

	constructor(id: number | undefined, name: string | undefined, description: string | undefined, user: User | undefined, created: Date | undefined, version: number | undefined) {
		this.id = id;
		this.name = name;
		this.description = description;
		this.user = user;
		this.created = created;
		this.version = version;
	}
	id: number | undefined;
	name: string | undefined;
	description: string | undefined;
	user: User | undefined;
	created: Date | undefined;
	version: number | undefined;
}

export class User {

	constructor(id: number | undefined, name: string | undefined, email: string | undefined, password: string | undefined, created: Date | undefined, version: number | undefined) {
		this.id = id;
		this.name = name;
		this.email = email;
		this.password = password;
		this.created = created;
		this.version = version;
	}
	id: number | undefined;
	name: string | undefined;
	email: string | undefined;
	password: string | undefined;
	created: Date | undefined;
	version: number | undefined;
}
